src/iphymv.cpp

```diff
@@ -73,53 +73,97 @@ main( int argc, char **argv ) {
+            // The checksum on the destination replica is wrong, indicating corruption.
+            return 4;
+
+        case USER_INPUT_PATH_ERR:
+            // The specified logical path does not exist, indicating a user error.
+            return 5;
+
+        case UNIX_FILE_CREATE_ERR:
+        case UNIX_FILE_OPEN_ERR:
+        case UNIX_FILE_WRITE_ERR:
+            // Return a special error code if there is no space left on the device.
+            // All other UNIX filesystem-related errors get a separate, but still special error code.
+            return ENOSPC == getErrno(status) ? 6 : 7;
+
+        default:
+            break;
     }
 
-}
+    // For any other errors, return 3 as has been done historically for minimal surprises.
+    return status < 0 ? 3 : 0;
+} // main
 
 void
 usage() {
 
     char *msgs[] = {
-        "Usage: iphymv [-hMrvV] [-n replNum] [-S srcResource]  [-R destResource] ",
+        "Usage: iphymv [-hMrvV] [-n replNum] [-S srcResource] [-R destResource] ",
         "dataObj|collection ... ",
         " ",
         "Physically move a file in iRODS to another storage resource.",
         " ",
+        "The source replica must be specified by its full hierarchy and it must",
+        "exist at the provided resource hierarchy location. The resource hierarchy",
+        "for the destination replica must also be specified in full, but may or may",
+        "not exist at the specified location. Specifying a leaf resource is acceptable",
+        "in lieu of a full resource hierarchy as it can be reverse-resolved to the root.",
+        "See these example usages:",
+        "  iphymv -S demoResc -R 'pt1;r1;s1' /tempZone/home/alice/test.txt",
+        "  iphymv -S demoResc -R 's1' /tempZone/home/alice/test.txt",
+        " ",
+        "If the leaf resource targeted for a phymv refers to the archive of a compound",
+        "resource hieararchy, the DIRECT_ARCHIVE_ACCESS will be returned because",
+        "direct access to an archive resource is not allowed.",
+        " ",
+        "iphymv cannot overwrite a replica which is not marked stale.",
+        "iphymv cannot move a replica which is not at rest.",
+        "iphymv cannot move a replica which is part of an already locked data object.",
+        " ",
+        "iphymv returns a few unique error codes based on the type of error which",
+        "occurred within the system:",
+        "  0: Success",
+        "  1: There was a problem with the client configuration or parsing the user input",
+        "  2: There was a problem connecting to the iRODS server (rcConnect failed)",
+        "  3: Historical exit code returned for all errors other than those listed above and below",
+        "  4: There was a mismatch between the source and destination replica's checksums",
+        "  5: There was a problem with the supplied source logical path",
+        "  6: The resource to host the destination replica has run out of disk space (ENOSPC)",
```

Comment:
may not be a disk.. can just say 'run out of space'

Reply:
Done